src/main/java/gov/nasa/cumulus/CnmToGranuleHandler.java

```diff
@@ -158,7 +165,7 @@ public String PerformFunction(String input, Context context) throws Exception {
 				
 		for (JsonElement file: inputFiles) {
 			JsonObject cnmFile = file.getAsJsonObject();
-
+			boolean isCNMResponseFileObject = this.containProviderDownloadUri(cnmFile, distribution_endpoint);
```

Comment:
Does this check need to happen for every file? Looking at a sample CNM response such as https://github.com/podaac/cloud-notification-message-schema/blob/master/samples/cumulus_sns_v1.5_response_success_product.json#L35 this check can be simplified by just checking if the response object exists and this can be done right before the for loop.